source/common/http/http1/codec_impl.cc

```diff
@@ -1115,6 +1139,27 @@ int ClientConnectionImpl::onHeadersComplete() {
       }
     }
 
+    // https://tools.ietf.org/html/rfc7230#section-3.3.3
+    // If a message is received with both a Transfer-Encoding and a
+    // Content-Length header field, the Transfer-Encoding overrides the
+    // Content-Length. Such a message might indicate an attempt to
+    // perform request smuggling (Section 9.5) or response splitting
+    // (Section 9.4) and ought to be handled as an error. A sender MUST
+    // remove the received Content-Length field prior to forwarding such
+    // a message.
+
+    // Reject response if Transfer-Encoding and Content-Length headers are present
+    // or remove Content-Length and serve response if allowed by http1 codec settings.
```

Comment:
This entire block and (and code below) is duplicated in the server case. Can we share a function that checks and handles it? The error case is slightly different but that should be easy to happen once the return status is checked from the shared funtion.

Reply:
It may be possible to do this error check in onHeadersCompleteBase.

Reply:
moved all validation logic to onHeadersCompleteBase(), code is not duplicated.